@codex 

apps/ocpp/services/certificate_status.py:9: error: Library stubs not installed
for "requests.adapters"  [import-untyped]
    from requests.adapters import HTTPAdapter
    ^
apps/ocpp/services/certificate_status.py:9: note: Hint: "python3 -m pip install types-requests"
apps/ocpp/services/certificate_status.py:9: note: (or run "mypy --install-types" to install all missing stub packages)
apps/ocpp/services/certificate_status.py:9: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 39 source files)

--- a/apps/repos/services/github.py
+++ b/apps/repos/services/github.py
@@ -55,6 +55,9 @@ JSONScalar: TypeAlias = str | int | float | bool | None
 JSONMapping: TypeAlias = dict[str, Any]
 JSONList: TypeAlias = list[Any]
 JSONValue: TypeAlias = JSONMapping | JSONList | JSONScalar
+RequestParamScalar: TypeAlias = str | bytes | int | float
+RequestParamValue: TypeAlias = RequestParamScalar | Iterable[RequestParamScalar] | None
+RequestParams: TypeAlias = Mapping[str, RequestParamValue]
 
 
 def build_headers(token: str, *, user_agent: str = "arthexis-admin") -> Mapping[str, str]:
@@ -238,12 +241,12 @@ def fetch_paginated_items(
     *,
     token: str,
     endpoint: str,
-    params: Mapping[str, object],
+    params: RequestParams,
     timeout: int = REQUEST_TIMEOUT,
 ) -> Iterator[Mapping[str, object]]:
     headers = build_headers(token)
     url = endpoint
-    query_params: Mapping[str, object] | None = params
+    query_params: RequestParams | None = params
 
     while url:
         response = None
@@ -284,7 +287,7 @@ def fetch_repository_issues(
     state: str = "open",
 ) -> Iterator[Mapping[str, object]]:
     endpoint = f"{API_ROOT}/repos/{owner}/{name}/issues"
-    params = {"state": state, "per_page": 100}
+    params: dict[str, RequestParamValue] = {"state": state, "per_page": 100}
     yield from fetch_paginated_items(token=token, endpoint=endpoint, params=params)
 
 
@@ -296,7 +299,7 @@ def fetch_repository_pull_requests(
     state: str = "open",
 ) -> Iterator[Mapping[str, object]]:
     endpoint = f"{API_ROOT}/repos/{owner}/{name}/pulls"
-    params = {"state": state, "per_page": 100}
+    params: dict[str, RequestParamValue] = {"state": state, "per_page": 100}
     yield from fetch_paginated_items(token=token, endpoint=endpoint, params=params)
 
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -102,6 +102,7 @@ ci = [
   "mypy==1.20.0",
   "ruff==0.15.9",
   "twine==6.1.0",
+  "types-requests==2.32.4.20250913",
 ]
 dev = [
   "django-debug-toolbar==6.3.0",

--- a/requirements-ci.txt
+++ b/requirements-ci.txt
@@ -87,6 +87,7 @@ trio==0.31.0
 twine==6.1.0
 Twisted==25.5.0
 txaio==25.9.2
+types-requests==2.32.4.20250913
 typing_extensions==4.15.0
 urllib3~=2.6.3
 webauthn==2.7.1

